e2e/taskevents/taskevents.go

```diff
@@ -0,0 +1,167 @@
+package taskevents
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/nomad/api"
+	"github.com/hashicorp/nomad/e2e/framework"
+	"github.com/hashicorp/nomad/testutil"
+	"github.com/stretchr/testify/require"
+
+	"github.com/hashicorp/nomad/e2e/e2eutil"
+	"github.com/hashicorp/nomad/helper/uuid"
+)
+
+type TaskEventsTest struct {
+	framework.TC
+	jobIds []string
+}
+
+func init() {
+	framework.AddSuites(&framework.TestSuite{
+		Component:   "TaskEvents",
+		CanRunLocal: true,
+		Cases: []framework.TestCase{
+			new(TaskEventsTest),
+		},
+	})
+}
+
+func (tc *TaskEventsTest) BeforeAll(f *framework.F) {
+	e2eutil.WaitForLeader(f.T(), tc.Nomad())
+	e2eutil.WaitForNodesReady(f.T(), tc.Nomad(), 1)
+}
+
+func (tc *TaskEventsTest) AfterEach(f *framework.F) {
+	nomadClient := tc.Nomad()
+	jobs := nomadClient.Jobs()
+	// Stop all jobs in test
+	for _, id := range tc.jobIds {
+		jobs.Deregister(id, true, nil)
+	}
+	// Garbage collect
+	nomadClient.System().GarbageCollect()
+}
+
+func formatEvents(events []*api.TaskEvent) string {
+	estrs := make([]string, len(events))
+	for i, e := range events {
+		estrs[i] = fmt.Sprintf("%2d %-20s fail=%t msg=> %s", i, e.Type, e.FailsTask, e.DisplayMessage)
+	}
+	return strings.Join(estrs, "\n")
+}
+
+// waitUntilEvents submits a job and then waits until the expected number of
+// events exist.
+//
+// The job name is used to load the job file from "input/${job}.nomad", and
+// events are only inspected for tasks named the same as the job.
+func (tc *TaskEventsTest) waitUntilEvents(f *framework.F, job string, numEvents int) *api.TaskState {
+	t := f.T()
+	nomadClient := tc.Nomad()
+	uuid := uuid.Generate()
+	jobId := "te_" + uuid[0:8]
```

Comment:
Don't really see how this job ID matches the one being registered?

Reply:
Ha, well that explains why it wasn't being cleaned up properly! Thanks.